Client/mods/deathmatch/logic/CLocalServer.cpp

```diff
@@ -198,18 +198,16 @@ bool CLocalServer::Load()
             m_pEditPlayers->SetText(pServerPlayers->GetTagContent().c_str());
 
         // Read the startup resources
-        list<CXMLNode*>::const_iterator iter = pRoot->ChildrenBegin();
-        for (; iter != pRoot->ChildrenEnd(); ++iter)
+        for (const auto& pNode : *pRoot)
         {
-            CXMLNode* pNode = reinterpret_cast<CXMLNode*>(*iter);
-            if (pNode->GetTagName().compare("resource") == 0)
-            {
-                CXMLAttribute* src = pNode->GetAttributes().Find("src");
-                if (src && src->GetValue()[1])
-                {
-                    m_pResourcesCur->SetItemText(m_pResourcesCur->AddRow(), m_hResourcesCur, src->GetValue().c_str());
-                }
-            }
+            if (pNode->GetTagName().compare("resource"))
+                continue;
+
+            CXMLAttribute* src = pNode->GetAttributes().Find("src");
+            if (!src || !src->GetValue()[1])
+                continue;
```

Comment:
Too many `continue` statements. In this particular case it makes things worse and absolutely redundant.
```
if (src && src->GetValue()[1])
    m_pResourcesCur->SetItemText(m_pResourcesCur->AddRow(), m_hResourcesCur, src->GetValue().c_str());
```
much better

Reply:
How does it makes things worse?
the line is already too big. And its easily changeable

Reply:
`continue` should not be used in everywhere. And it's better to avoid the negation in expressions. It's simply difficult to read.

Reply:
> `continue` should not be used in everywhere. And it's better to avoid the negation in expressions. It's simply difficult to read.

I beg to differ. Chained `if` statements are far worse than `continue` everywhere. Well, we can already see the results of it by a poor readability in current implementation.

Reply:
There is no nested statements in the example above. It's okay when you `continue` in the beginning of a cycle, but only in the beginning. 
```
{
        CXMLAttribute* src = pNode->GetAttributes().Find("src");
	if (!src || !src->GetValue()[1])
                continue;

	m_pResourcesCur->SetItemText(m_pResourcesCur->AddRow(), m_hResourcesCur, src->GetValue().c_str());
}
```
is defenitely the overuse of continue. It takes more code lines and makes it difficult to read the code.

Reply:
A bigger line doesn't increase readability. `continue` increases it by removing the additional tab there.
Negations in `if` statements are trivial. Trivial to read, write and understand. Though it should be refactored a little bit more than just removing nested `if`

Reply:
> A bigger line doesn't increase readability. `continue` increases it by removing the additional tab there. Negations in `if` statements are trivial. Trivial to read, write and understand. Though it should be refactored a little bit more than just removing nested `if`

There is no "bigger line". Only considerations of clean and readable code. 
P.S. How many programmers, so many opinions, for sure. But I bet that most of the programmers have the same opinion that I have. But I do not exclude the possibility of the wrongness of such opinion though.

Reply:
> There is no "bigger line". 

There literally is. A tab significantly increases line's length. And bigger lines are less readable.

> How many programmers, so many opinions, for sure. But I bet that most of the programmers have the same opinion that I have. But I do not exclude the possibility of the wrongness of such opinion though.
I agree with it. But I don't agree with removing `continue`

Reply:
imo it's fine either way